### ISSUE TITLE
Guideline for adding new examples

### ISSUE THREAD

**Issue by Aakash-kaushik** [mlpack/examples#120]
So this issue gives out a template or a guide that can be followed or kept in mind when writing new examples:

- The examples should be well documented in terms that can easily be understood by people entering the field of Machine learning or have been in it from a long time.

- Make examples simple enough so the user can basically grasp what he needs to build his own full fledged model and don't over-complicate the examples but still try to include all library functionalities related to that example.

- A little bit more can be written about parameters or functionalities which are implemented differently in mlpack than what the common notion is or which might confuse the user.

- And please take a look at the example that exist before and try to avoid redundant examples. 

And lastly this is just a guide not strict rules so have fun writing examples and show your creativity. **:)**

**Comment by shrit** [comment]
@Aakash-kaushik Thank you for writing this, I think we can add some points about tutorials too.
It is quite straightforward to write examples. Most of the above points can be agreed on by most of us.
However, it would interesting to add some points about comments and tutorials considering each example, users would like to know why we followed this pattern in writing code :+1:

**Comment by birm** [comment]
I think we should leave space for especially interesting or practical examples, even if they aren't especially simple.

**Comment by Aakash-kaushik** [comment]
> @Aakash-kaushik Thank you for writing this, I think we can add some points about tutorials too.
> It is quite straightforward to write examples. Most of the above points can be agreed on by most of us.
> However, it would interesting to add some points about comments and tutorials considering each example, users would like to know why we followed this pattern in writing code :+1: 

I need to think about how and what to write and then i will add it.

**Comment by Aakash-kaushik** [comment]
> I think we should leave space for especially interesting or practical examples, even if they aren't especially simple.

Yes that is correct I will add that point too.

**Comment by Aakash-kaushik** [comment]
Hey @shrit, @birm give it a look now.

**Comment by zoq** [comment]
@Aakash-kaushik Thanks for putting this together, can you run a spell checker over the text?

**Comment by Aakash-kaushik** [comment]
> @Aakash-kaushik Thanks for putting this together, can you run a spell checker over the text?

Edited the issue, please take a look.

**Comment by shrit** [comment]
@rcurtin do you think we can pin this issue?

**Comment by rcurtin** [comment]
Yeah!  Not a bad idea.  Done. :+1:  Thanks @Aakash-kaushik for originally writing these up!

**Comment by Saikirannaidu225** [comment]
I will add some example tutorial using mlpack. I will document the examples well understood and precise. Please assign this to me

**Comment by Aakash-kaushik** [comment]
> I will add some example tutorial using mlpack. I will document the examples well understood and precise. Please assign this to me

Hey @Saikirannaidu225, welcome and feel free to take those up and create a draft PR for those.
But this issue is just a skeleton and only serves as a guideline on how things should be and not as an actual issue.
